base/hashing.jl

```diff
@@ -17,90 +22,67 @@ The hash value may change when a new Julia process is started.
 
 ```jldoctest; filter = r"0x[0-9a-f]{16}"
 julia> a = hash(10)
-0x95ea2955abd45275
+0x759d18cc5346a65f
 
 julia> hash(10, a) # only use the output of another hash function as the second argument
-0xd42bad54a8575b16
+0x03158cd61b1b0bd1
 ```
 
 See also: [`objectid`](@ref), [`Dict`](@ref), [`Set`](@ref).
 """
-hash(x::Any) = hash(x, zero(UInt))
+hash(data::Any) = hash(data, HASH_SEED)
 hash(w::WeakRef, h::UInt) = hash(w.value, h)
 
 # Types can't be deleted, so marking as total allows the compiler to look up the hash
-hash(T::Type, h::UInt) = hash_uint(3h - @assume_effects :total ccall(:jl_type_hash, UInt, (Any,), T))
+hash(T::Type, h::UInt) =
+    hash((@assume_effects :total ccall(:jl_type_hash, UInt, (Any,), T)), h)
```

Comment:
Should this use hash_mix?

Reply:
that will expand to `hash_finalizer(hash_mix_linear(type_hash, h))` which is most directly analogous to the previous implementation, and matches the other methods on all the paths that hit the `UInt` hash (aka everything except strings).